Spigot-API-Patches/0276-Add-PlayerShieldBreakEvent.patch

```diff
@@ -0,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tom <cryptite@gmail.com>
+Date: Fri, 26 Feb 2021 17:16:17 -0600
+Subject: [PATCH] Add PlayerShieldBreakEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerShieldBreakEvent.java b/src/main/java/org/bukkit/event/player/PlayerShieldBreakEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ba97c78022e7ddd40d6e1c0b9bd983aaf3f96277
+--- /dev/null
++++ b/src/main/java/org/bukkit/event/player/PlayerShieldBreakEvent.java
+@@ -0,0 +1,44 @@
++package org.bukkit.event.player;
++
++import org.bukkit.entity.LivingEntity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Thrown whenever a {@link Player} dies
++ */
++public class PlayerShieldBreakEvent extends Event {
```

Comment:
A very personal opinion here that I simply want to propose:

1. This event could inherit from the [PlayerEvent](https://papermc.io/javadocs/paper/1.16/org/bukkit/event/player/PlayerEvent.html) parent class which is already used as a super class for most player related events. This would also reduce the diff of this patch :)
2. The naming of this event seems a bit questionable. A shield may in fact **break** and be completely removed, but the code implementation you provided here is throwing this event whenever the shield is **disabled by an axe**. The name "break" therefore seems, at least to me, a little off the actual event context (See [official shield wiki entry for naming convention](https://minecraft.gamepedia.com/Shield).

Reply:
> 
> 
> A very personal opinion here that I simply want to propose:
> 
>     1. This event could inherit from the [PlayerEvent](https://papermc.io/javadocs/paper/1.16/org/bukkit/event/player/PlayerEvent.html) parent class which is already used as a super class for most player related events. This would also reduce the diff of this patch :)
> 
>     2. The naming of this event seems a bit questionable. A shield may in fact **break** and be completely removed, but the code implementation you provided here is throwing this event whenever the shield is **disabled by an axe**. The name "break" therefore seems, at least to me, a little off the actual event context (See [official shield wiki entry for naming convention](https://minecraft.gamepedia.com/Shield).

2 - This event specifically should be renamed to "PlayerShieldDisabledEvent" and a new "PlayerShieldBreakEvent" event added for when it ACTUALLY breaks

Reply:
`PlayerItemBreakEvent` already exists.